service/matching/version_sets_test.go

```diff
@@ -644,7 +680,68 @@ func TestGetBuildIdDeltas(t *testing.T) {
 }
 
 func TestGetBuildIdDeltas_AcceptsNils(t *testing.T) {
+	t.Parallel()
 	added, removed := GetBuildIdDeltas(nil, nil)
 	assert.Equal(t, []string(nil), removed)
 	assert.Equal(t, []string(nil), added)
 }
+
+func TestMergeSets(t *testing.T) {
+	t.Parallel()
+	clock := hlc.Zero(1)
+	initialData := mkInitialData(4, clock)
+
+	req := mkMergeSet("1", "2")
+	nextClock := hlc.Next(clock, commonclock.NewRealTimeSource())
+	updatedData, err := UpdateVersionSets(nextClock, initialData, req, 0, 0)
+	assert.NoError(t, err)
+	// Should only be three sets now
+	assert.Equal(t, 3, len(updatedData.VersionSets))
+	// The overall default set should not have changed
+	assert.Equal(t, "3", updatedData.GetVersionSets()[2].GetBuildIds()[0].Id)
+	// But set 1 should now have 2, maintaining 1 as the default ID
+	assert.Equal(t, "1", updatedData.GetVersionSets()[1].GetBuildIds()[1].Id)
```

Comment:
can you verify that it has two set ids and they match the individual set ids from before?

Reply:
Added